src/gesture-handler/GestureDetector.tsx

```diff
@@ -0,0 +1,302 @@
+  const startingGesturePosition = useSharedValue(DefaultEvent);
+  const canPerformUpdates = useSharedValue(false);
+  let transitionAnimation = ScreenTransition.SwipeRight;
+  if (userTransitionAnimation) {
+    transitionAnimation = userTransitionAnimation;
+    if (!goBackGesture) {
+      throw new Error(
+        '[RNScreens] You have to specify `goBackGesture` when using `transitionAnimation`'
+      );
+    }
+  } else {
+    if (!!goBackGesture && SupportedGestures.includes(goBackGesture)) {
+      transitionAnimation = AnimationForGesture[goBackGesture];
+    } else if (goBackGesture !== undefined) {
+      throw new Error(`[RNScreens] Unknown goBackGesture: ${goBackGesture}`);
+    }
+  }
+  const screenTransitionConfig = makeMutable(
+    {
+      stackTag: -1,
+      belowTopScreenTag: -1,
+      topScreenTag: -1,
+      sharedEvent,
+      startingGesturePosition,
+      screenTransition: transitionAnimation,
+      isTransitionCanceled: false,
+      goBackGesture: goBackGesture ?? 'swipeRight',
+      screenDimensions: DefaultScreenDimensions,
+      onFinishAnimation: () => {
+        'worklet';
+      },
+    },
+    true
+  );
+  const refCurrent = stackRef.current;
+  if (refCurrent === null) {
+    throw new Error('[RNScreens] You have to specify `stackRefWrapper`');
+  }
+
+  function onStart(event: GestureUpdateEvent<PanGestureHandlerEventPayload>) {
+    'worklet';
+    sharedEvent.value = event;
+    const transitionConfig = screenTransitionConfig.value;
+    const animatedRef = refCurrent;
+    if (!animatedRef) {
+      throw new Error('[Reanimated] Unable to recognize stack ref.');
+    }
+    const stackTag = (animatedRef as () => number)();
+    const screenTags = global._manageScreenTransition(
+      ScreenTransitionCommand.Start,
+      stackTag,
+      null
+    );
+    if (!screenTags) {
+      canPerformUpdates.value = false;
+      return;
+    }
+    transitionConfig.topScreenTag = screenTags.topScreenTag;
+    transitionConfig.belowTopScreenTag = screenTags.belowTopScreenTag;
+    transitionConfig.stackTag = stackTag;
+    startingGesturePosition.value = event;
+    const animatedRefMock = () => {
+      return screenTransitionConfig.value.topScreenTag;
+    };
+    const screenSize = measure(animatedRefMock as any);
+    if (screenSize == null) {
+      throw new Error('[Reanimated] Failed to measure screen.');
+    }
+    if (screenSize == null) {
+      canPerformUpdates.value = false;
+      global._manageScreenTransition(
+        ScreenTransitionCommand.Finish,
+        stackTag,
+        true
+      );
+      return;
+    }
+    transitionConfig.screenDimensions = screenSize;
+    startScreenTransition(transitionConfig);
+    canPerformUpdates.value = true;
+  }
+
+  function onUpdate(event: GestureUpdateEvent<PanGestureHandlerEventPayload>) {
+    'worklet';
+    if (!canPerformUpdates.value) {
+      return;
+    }
+    if (goBackGesture === 'swipeRight' && event.translationX < 0) {
```

Comment:
We need to change all those numbers etc into descriptive variables, or maybe create small functions like `resetNegativeTranslation` here?